app/src/main/java/com/tari/android/wallet/ui/activity/transactiondetails/TransactionDetailActivity.kt

```diff
@@ -0,0 +1,254 @@
+    @BindView(R.id.tx_detail_txt_tx_fee)
+    lateinit var txFeeTextView: CustomFontTextView
+
+    @BindView(R.id.tx_detail_btn_add_contact)
+    lateinit var addContactButton: CustomFontButton
+
+    @BindView(R.id.tx_detail_txt_contact_name)
+    lateinit var contactNameTextView: CustomFontTextView
+
+    @BindView(R.id.tx_detail_edit_create_contact)
+    lateinit var contactEditText: CustomFontEditText
+    @BindView(R.id.tx_detail_txt_contact_label)
+    lateinit var contactLabelTextView: CustomFontTextView
+
+    @BindView(R.id.tx_detail_txt_edit_label)
+    lateinit var editContactLabelTextView: CustomFontTextView
+
+    @BindView(R.id.tx_detail_txt_tx_note)
+    lateinit var transactionNoteTv: CustomFontTextView
+
+    @BindView(R.id.tx_detail_txt_tx_id)
+    lateinit var txIdTextView: CustomFontTextView
+
+    @BindView(R.id.tx_detail_vw_contact_container)
+    lateinit var contactContainerView: View
+
+    @BindView(R.id.tx_detail_vw_emoji_summary)
+    lateinit var emojiSummaryView: View
+    @BindView(R.id.tx_detail_vw_separator)
+    lateinit var contactSeparatorView: View
+
+    @BindView(R.id.tx_detail_txt_note_label)
+    lateinit var noteLabelView: View
+
+    @BindView(R.id.tx_detail_vw_tx_fee_group)
+    lateinit var txFeeGroup: Group
+
+    private var transaction: Tx? = null
+    private lateinit var emojiIdSummaryController: EmojiIdSummaryViewController
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        overridePendingTransition(R.anim.enter_from_right, android.R.anim.fade_out)
+        emojiIdSummaryController = EmojiIdSummaryViewController(emojiSummaryView)
+        transaction = intent.getParcelableExtra(TX_DETAIL_EXTRA_KEY)
+        if (transaction == null) finish()
+        setupUI()
+    }
+
+    override fun onBackPressed() {
+        super.onBackPressed()
+        overridePendingTransition(R.anim.enter_from_left, R.anim.exit_to_right)
+    }
+
+    @SuppressLint("SetTextI18n")
+    private fun setupUI() {
+        when (transaction) {
+            is CompletedTx -> {
+                when (transaction!!.direction) {
+                    Tx.Direction.INBOUND -> txPaymentStateTextView.text =
+                        getString(R.string.tx_detail_payment_received)
+                    Tx.Direction.OUTBOUND -> txPaymentStateTextView.text =
+                        getString(R.string.tx_detail_payment_sent)
+                }
+            }
+            is PendingInboundTx -> txPaymentStateTextView.text =
+                getString(R.string.tx_detail_pending_payment_received)
+            is PendingOutboundTx -> txPaymentStateTextView.text =
+                getString(R.string.tx_detail_pending_payment_sent)
+        }
+
+        val timeStamp = transaction!!.timestamp.toLong() * 1000
+        txDateTextView.text = DateUtils.getTxFormattedDate(Date(timeStamp))
+        txAmountTextView.text = "%1$,.2f".format(transaction!!.amount.tariValue.toDouble())
```

Comment:
See the formatters in `WalletUtil`, one of them may help - and you can pass `BigDecimal` to formatters, examples in other classes.